### PR TITLE
c-s: counters

### PR DESCRIPTION
**Pull request by muzarski**
# Changes
Implemented runtime for both `counter_read` and `counter_write` commands.

This includes:
- renaming `ReadWriteParams` to `CommonParams`
- parsing additional parameter for `counter_write` command (`add=`)
- making `ReadOperation` generic over `RowValidator` trait, since `counter_read` differs from `read` only with validation logic
- implemented `CounterWriteOperation`
- implemented `CounterReadOperation` (as `ReadOperation<ExistsRowValidator>`)

### CODE REVIEW

**Review by piodul — CHANGES_REQUESTED** [review]

File: src/bin/cql-stress-cassandra-stress/settings/param/parser.rs [hunk 1]
```diff
@@ -59,6 +59,10 @@ impl ParamsGroup {
         true
     }
 
+    fn append(&mut self, params: &[ParamCell]) {
```
[the remark this answers is not in the document]
Comment (muzarski) [reply src/bin/cql-stress-cassandra-stress/settings/param/parser.rs:62]: Actually, I just realized that c-s doesn't distinguish common (`read/write`) and counter workloads' parameters. It accepts something like:
```
./cassandra-stress  write n=10 'add=FIXED(10)'
```
Obviously, in this case, the `add=` will be simply ignored since it's only used during `counter_write` workload. What do you think about following the c-s approach and just extending the existing `CommonParams` (the name `common` still holds, because `user` workload uses common parameters + some params specific only to itself) workload structure by `add_distribution`?

BTW, what is strange is that `add=` parameter is not printed in the help message:
```
./cassandra-stress help counter_write
Usage: counter_write [err<?] [n>?] [n<?] [no-warmup] [truncate=?] [cl=?] [serial-cl=?]
 OR 
Usage: counter_write n=? [no-warmup] [truncate=?] [cl=?] [serial-cl=?]
 OR 
Usage: counter_write duration=? [no-warmup] [truncate=?] [cl=?] [serial-cl=?]

  err<? (default=0.02)                     Run until the standard error of the mean is below this fraction
  n>? (default=30)                         Run at least this many iterations before accepting uncertainty convergence
  n<? (default=200)                        Run at most this many iterations before accepting uncertainty convergence
  no-warmup                                Do not warmup the process
  truncate=? (default=never)               Truncate the table: never, before performing any work, or before each iteration
  cl=? (default=LOCAL_ONE)                 Consistency level to use
  serial-cl=? (default=SERIAL)             Serial consistency level to use
  n=?                                      Number of operations to perform
  duration=?                               Time to run in (in seconds, minutes or hours)
```
And what is even more strange, is that apart from `add=` parameter, there is also `keysize=` parameter which is as well accepted (and used in runtime), but not printed in the help message.

So to summarize - I think we should just extend `CommonParams` by `add=` and `keysize=` parameters.

**Comment by muzarski** [comment]
v2:
- fixed review comments
- extended `CommonParams` by `keysize` and `add_distribution` parameters (instead of introducing `CounterParams`)

File: src/bin/cql-stress-cassandra-stress/settings/param/parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (piodul) [reply src/bin/cql-stress-cassandra-stress/settings/param/parser.rs]: I think it would be better not to accept parameters if they are going to be ignored. If a user types

```
./cassandra-stress  write n=10 'add=FIXED(10)'
```

they might expect the `add=` parameter to have some effect, but it will be ignored which is not what the user wants. I think it's completely fine not to be compatible here.

**Comment by muzarski** [comment]
v3:
- rebased on top of master
- fixed review comments - changed the parsing logic so 'add=' parameter is accepted only for the `counter_write` workload.

**Review by piodul — APPROVED** [review]
